test/integration/issuance_test.go

```diff
@@ -167,3 +170,69 @@ func TestIssuanceProfiles(t *testing.T) {
 	test.AssertEquals(t, len(legacy.SubjectKeyId), 20)
 	test.AssertEquals(t, len(modern.SubjectKeyId), 0)
 }
+
+// TestIPShortLived verifies that we will allow IP address identifiers only in
+// orders that use the shortlived profile.
+func TestIPShortLived(t *testing.T) {
+	t.Parallel()
+
+	// Create an account.
+	client, err := makeClient("mailto:example@letsencrypt.org")
+	if err != nil {
+		t.Fatalf("creating acme client: %s", err)
+	}
+
+	_, shortlived := client.Directory().Meta.Profiles["shortlived"]
+	if !shortlived {
+		t.Fatal("ACME server not advertising the shortlived profile")
+	}
+
+	// Create a private key.
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		t.Fatalf("creating random cert key: %s", err)
+	}
+
+	// Create an IP address identifier to request.
+	idents := []acme.Identifier{
+		{Type: "ip", Value: "64.112.117.122"},
+	}
+
+	// Ensure we fail under each other profile that we know the test server advertises.
+	_, err = authAndIssue(client, key, idents, false, "legacy")
+	if err == nil {
+		t.Fatal("issued for IP address identifier under legacy profile")
```

Comment:
a) This should just be Error, not Fatal, because the rest of the test can continue just fine from here
b) This should be followed by a second `if !strings.Contains(err.Error(), "some specific error message")` to confirm that we got rejected due to the IP identifier, and not for some other reason (e.g. specifying an unrecognized profile).